src/lib/config/GeocodingConfig.ts

```diff
@@ -1,27 +1,100 @@
-import {
-  GeocodingService,
-} from "$lib/services/geocoding";
-import { dev } from "$app/environment";
-
-// Configuration for different environments
-const config = {
-  development: {
-    provider: "nominatim",
-    endpoint: "https://nominatim.openstreetmap.org",
-    userAgent: "YourApp/1.0 (Development)",
-    email: "your-email@example.com",
-    rateLimit: 1000,
-  },
-  production: {
-    provider: "nominatim",
-    endpoint: "https://nominatim.openstreetmap.org",
-    userAgent: "YourApp/1.0 (Production)",
-    email: "your-email@example.com",
-    rateLimit: 1000,
-  },
-} as const;
-
-// Export the configured service
-export const geocodingService = new GeocodingService(
-  dev ? config.development : config.production
-);
\ No newline at end of file
+import axios from "axios";
+
+// Define interface for geocoding response
+interface GeocodingResponse {
+   valid: boolean;
+   lat?: number;
+   lng?: number;
+   error?: string;
+   formattedAddress?: string;
+}
+
+export const geocodingService = {
+   geocode: async (
+      address: string,
+      city: string,
+      country: string,
+      postal: string
+   ): Promise<GeocodingResponse> => {
+      let fullAddress = `${address} ${city} ${country} ${postal}`;
+      fullAddress = encodeURIComponent(fullAddress.trim());
+
+
+      console.log("Geocoding address:", fullAddress);
+
+      try {
+         const response = await axios.get(
+            "https://maps.googleapis.com/maps/api/geocode/json",
```

Comment:
ideally we would have used the official Google sdk, less-ideally we could have used existing dependencies or builtins to make http requests, but it will be ok